opflexagent/test/test_agent_types.py

```diff
@@ -38,7 +39,19 @@ def test_opflex_agent_mode(self):
             'opflexagent.as_metadata_manager.AsMetadataManager')
         metadata_patch = metadata_mgr.start()
         cfg.CONF.set_override('agent_mode', 'opflex', 'OPFLEX')
-        with mock.patch('sys.argv'):
+
+        resources = [
+            mock.patch('os.path.basename'),
+            mock.patch('sys.argv')]
+
+        @contextmanager
+        def nested(*contexts):
```

Comment:
this nested() API here is being defined multiple times in each individual test function in this and other test classes. We should define it as a function in some base test class then each test API can just invoke it directly.

Reply:
Yeah, it makes sense to define it as a function.